### PR TITLE
Initial BlockDeleter for #3080. DOES NOT work on mutliple cores. Leav…

### PR DESCRIPTION
**Pull request by ProfessorHoover**
…es hanging nodes, but physics still propogate correctly as intended, ignoring removed nodes

### CODE REVIEW

**Comment by WilkAndy** [comment]
It would be great if someone could help the Professor with this.  The problems he's identified are:

(1) When you delete elements in a block, some nodes remain (probably the nodeset ones - i'm not sure).

(2) When you try to run his test file on 2 processors, a segfault occurs (as i recall).  Maybe he should be deleting elements only if they are owned by a processor?

a

**Comment by permcody** [comment]
Thanks! You may have caught us at a bad time. SuperComputing is next week and most of the team is going to that. We have an audit next week as well so between the two, very little is going to happen before Thanksgiving.

As far as multiprocessor runs go: When using serial mesh (which you are by default and should use) every processor has a complete copy of the mesh and that copy needs to be consistent among all processors so deleting the same elements on all processors is what you want to do. The bigger problem I see is the removal of all the dependency resolution logic! That code is absolutely necessary. 

In a nutshell when deleting elements you have to delete them from the outside in. This might also be the issue with your hanging nodes. The way I accomplished that was by putting the elements into a structure that made sure whatever element was to be deleted next was on the outside of the current mesh. I can explain more if needed but that code was in the original deleter and it needs to remain there.

**Comment by permcody** [comment]
@ProfessorHoover, this is minor but helps with readability of commits. Try keeping your first line of your subject line short so it fits in the header nicely. Then put a blank line before the remaining part of your message and reference.

Commit Message Format:
```
Short Title

Longer description of your changes that can span multiple lines 
if necessary. You might even add bullets
 - feature one
 - feature two
 - fix one
closes #nnn
```

BTW: I don't think we've announced this convention but I'll add it to our wiki.

**Comment by permcody** [comment]
It dawned on me that you might need to delete elements out of the middle of the mesh where there aren't any boundaries. I haven't done that at all yet. I may have to take a look at this a bit more.

**Comment by WilkAndy** [comment]
You are correct, @permcody : sometimes we will just want to delete elements that aren't otherwise "interesting".

**Comment by YaqiWang** [comment]
I think it's proper to create a side set which contains the intersection of all internal sides and the surrounding sides of the block to be deleted. Of course all nodes, lines if there is any, sides, cells only on the block need to be deleted.

**Comment by permcody** [comment]
True, but first step is first, this code needs to work just deleting the elements.

**Comment by WilkAndy** [comment]
General request: - i'm on holidays until christmas although will be checking in with the Professor periodically.  I'd really appreciate it if you guys could provide help when he needs it.  I know you will, but wanted to let you know that he won't have much support from me over the next few weeks.

Sidesets: - usually i would have made sidesets in the original mesh, so although Yaqi's idea would be nice for other people, i'd probably not often use it.

**Comment by permcody** [comment]
When I get a chance, I'll check out this branch and run this problem. @ProfessorHoover, the next steps will be to figure out where the crash is happening. Have you compiled in debug mode yet and run through the debugger?

**Comment by permcody** [comment]
Merged into #6081
